pacman/utilities/utility_objs/resource_tracker.py

```diff
@@ -1288,6 +1299,11 @@ def unallocate_resources(self, chip_x, chip_y, processor_id, resources,
         self._sdram_tracker -= resources.sdram.get_value()
         self._core_tracker[(chip_x, chip_y)].add(processor_id)
 
+        # check if chip used needs updating
+        if (len(self._core_tracker[(chip_x, chip_y)]) ==
+                self._machine.get_chip_at(chip_x, chip_y).n_processors):
+            self._chips_used.remove((chip_x, chip_y))
```

Comment:
Not sure this check is quite right.  Presumably you only want to remove it from chips_used if no processors are now used?  A test for what happens in various situations is really needed here...